### PR TITLE
Reduce from 10 hosts to 4 (bastion, master, infra, node)

### PR DESCRIPTION
**Pull request by jim-minter**

### CODE REVIEW

**Comment by jim-minter** [comment]
Currently testing the deployment

**Comment by jim-minter** [comment]
Deployment tested out fine :)

**Comment by kargakis** [comment]
@kwoodson

File: parts/openshift/3.7/bastion.sh [hunk 1]
```diff
@@ -0,0 +1,1341 @@
...
+      value:
+      - "{{ azure_conf }}"
+
+    - key: kubernetesMasterConfig.controllerArguments.cloud-provider
+      value:
+      - azure
+    notify:
+    - restart atomic-openshift-master-api
+    - restart atomic-openshift-master-controllers
+#
```
Comment (kargakis) [inline parts/openshift/3.7/bastion.sh]: cruft

File: parts/openshift/3.7/bastion.sh [hunk 2]
```diff
@@ -0,0 +1,1341 @@
...
+      msg: "Idm dir {{ idm_dir }}"
+  - name: Install Java
+    yum:
+      name: java-1.8.0-openjdk
+      state: latest
+  - name: Cleanup old idm directory
+    file:
+      state: absent
+      path: "{{idm_dir}}"
+  - name: C eate new idm directory
```
Comment (kargakis) [inline parts/openshift/3.7/bastion.sh]: missing char

File: parts/openshift/3.7/bastion.sh [hunk 3]
```diff
@@ -0,0 +1,1341 @@
...
+         SSO_TRUSTSTORE_PASSWORD="{{idm_xpassword}}"
+
+  - name: Stage 12 - OCCREATE SECRET ADD
+    command: oc new-app sso71-postgresql --param-file {{idm_dir}}/sso.params -l app=sso71-postgresql -l application=sso -l template=sso71-https
+  - name: Stage 13 - add xml pv
+    command: oc volume dc/sso --add --claim-size 512M --type=emptyDir --mount-path /opt/eap/standalone/configuration/standalone_xml_history --name standalone-xml-history
+  - set_fact: sso_token_url="https://login.{{sso_domain}}/auth/realms/cloud/protocol/openid-connect/token"
+  - name: Pause for app create
+    pause:
+      minutes: 4
```
Comment (kargakis) [inline parts/openshift/3.7/bastion.sh]: Ok, I guess if we are not going to use the bastion host eventually, it's not worth fixing these but you could simply use `oc rollout status dc/sso` here. @tnozicka did we fix the `oc rollout status` broken watch issue?

File: parts/openshift/3.7/bastion.sh [hunk 4]
```diff
@@ -0,0 +1,1341 @@
...
+    ;;
+esac
+
+BZ1469358
+
+echo "Done"
+EOF
+chmod a+x /home/${AUSERNAME}/add_host.sh
+
+npm install -g azure-cli
```
Comment (kargakis) [inline parts/openshift/3.7/bastion.sh]: I am wondering why npm over yum

https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-yum?view=azure-cli-latest

**Comment by kargakis** [comment]
@jim-minter if these scripts serve as a testbed for spinning up clusters until we have golden images, feel free to ignore my comments.

**Comment by pweil-** [comment]
deployed fine for me (minus some storage account conflicts due to me cancelling and redploying multiple times).

LGTM if we aren't changing any of the comments from @kargakis

**Comment by jim-minter** [comment]
@pweil- @kargakis' comments are all valid, but this is a dumb copy/paste trying to avoid changing as much as possible - please merge

File: parts/openshift/3.7/bastion.sh (hunk 3 above)
Comment (tnozicka) [reply parts/openshift/3.7/bastion.sh]: not yet, I need to get back to that PR when possible

Comment (tnozicka) [reply parts/openshift/3.7/bastion.sh]: @kargakis https://github.com/kubernetes/kubernetes/pull/50102

**Pull request merged**